plugins/Diagnostics/Commands/Run.php

```diff
@@ -73,6 +84,56 @@ protected function execute(InputInterface $input, OutputInterface $output)
         return 0;
     }
 
+    /**
+     * Handle unexpected files command options
+     *
+     * @param OutputInterface $output
+     * @param bool $delete
+     *
+     * @return int
+     */
+    private function runUnexpectedFiles(OutputInterface $output, bool $delete = false): int
+    {
+
+        if ($delete) {
+            $output->writeln('!!! Deleting all unexpected files in the Matomo directory - Press CTRL-C now to abort !!!', OutputInterface::OUTPUT_NORMAL);
+        }
+
+        // A list of file that should never be deleted under any circumstances, this acts as a backup safety check
+        // for the FileIntegrity class which should already be excluding these files.
+        $excludedFiles = ['config.ini.php', 'common.config.ini.php', '.htaccess', 'config.php', 'misc/'];
```

Comment:
Those checks might be a bit too loose. As they are currently checked with a `strpos` only, it might happen to match false positives.
If we e.g. would have and remove a file like `plugin/AnyName/Randomconfig.php`, it would be matched by `config.php`.
Might maybe be better to use regexes to match the excludes.

Reply:
Good point :+1: